org-node/src/client.rs

```diff
@@ -149,16 +168,41 @@ pub struct Head {
 
 /// Project peer contains details used to track the peer for an interested project Urn.
 /// this struct is primarily used by `TrackPeerTransmitter`
-struct TrackPeer {
-    api: Peer<Signer>,
-    peer_info: PeerInfo<std::net::SocketAddr>,
-    paths: Paths,
-    urn: Urn,
+pub struct TrackPeer {
+    pub api: Peer<Signer>,
+    pub peer_info: PeerInfo<std::net::SocketAddr>,
+    pub paths: Paths,
+    pub urn: Urn,
+}
+
+impl ToString for TrackPeer {
+    fn to_string(&self) -> String {
+        format!("{}:{}", self.peer_info.peer_id, self.urn)
+    }
 }
 
-/// Project peer struct with a acknowledgement transmitter.
+/// Track peer struct with a acknowledgement transmitter.
 /// This struct is primarily used by `protocol_listener` and TrackProject request handler.
-type TrackPeerTransmitter = (TrackPeer, oneshot::Sender<Result<Option<PeerId>, Error>>);
+type TrackPeerTransmitter = (
+    TrackPeer,
+    Option<oneshot::Sender<Result<Option<PeerId>, Error>>>,
+);
+
+/// General Message Bus for cross-thread communications.
+pub enum MessageBus {
+    TrackPeer(TrackPeerTransmitter),
+    /// Used to inform the message bus a new web socket connection has been established.
+    WebSocketConnection(mpsc::UnboundedSender<String>),
+}
+
+impl ToString for MessageBus {
```

Comment:
Would be good to discuss how we want to serialize messages; I don't think we can use serde here unless we implement the serialization ourselves, which I am not opposed to doing, just want to think through how the format of the messages should be responded back to client.